relay-server/src/utils/rate_limits.rs

```diff
@@ -378,9 +425,29 @@ where
         let mut enforcement = Enforcement::default();
 
         if let Some(category) = summary.event_category {
-            let event_limits = (self.check)(scoping.item(category), 1)?;
-            let longest = event_limits.longest();
-            enforcement.event = CategoryLimit::new(category, 1, longest);
+            let mut longest;
+            let mut event_limits;
+
+            if let Some(index_category) = self.index_category(category) {
+                // Check for rate limits on the main category (e.g. transaction) but do not consume
+                // quota. Quota will be consumed by metrics in the metrics aggregator instead.
+                event_limits = (self.check)(scoping.item(category), 0)?;
+                longest = event_limits.longest();
+                enforcement.event_metrics = CategoryLimit::new(category, 1, longest);
```

Comment:
This is basically it. The original idea was to create a non-optional type that we assign unconditionally, and that carries all information about the enforced rate limit. The only exposed method ended being `is_active`, which checks if we have a non-zero quantity. It's factually correct here, since we are in fact limiting `1` transaction.